### PR TITLE
Update user-level documentation workflow

### LINKED ISSUE #1896: Update documentation workflow and doc/README.rst
The user-level documentation instructions need to be updated to reflect the changes in #1843.

To avoid duplication, the relevant sections in ``doc/README.rst`` can be replaced with links to the [User level documentation workflow](https://github.com/nest/nest-simulator/blob/master/doc/documentation_workflow/user_documentation_workflow.rst) and [Developer documentation workflow](https://github.com/nest/nest-simulator/blob/master/doc/documentation_workflow/developer_documentation_workflow.rst).

### PR DESCRIPTION
**Pull request by sarakonradi**
This PR updates the user-level documentation workflow to reflect the changes in #1843. I have numbered the instructions for generating HTML files with Sphinx and added a step on how to preview the documentation locally.

It fixes #1896.

### CODE REVIEW

File: doc/userdoc/documentation_workflow/user_documentation_workflow.rst [hunk 1]
```diff
@@ -80,17 +80,30 @@ Sphinx.
...
 ~~~~~~~~~~~~~~
 
-If you intend to work on the documentation, or if you want to obtain a
-local version of it for other reasons, you can build the documentation
-by simply running the following command in the build directory of NEST
-(i.e. the directory where you ran ``cmake``)
+You can build and preview the documentation locally by running the following
+commands.
+
+1. Go to the :ref:`build directory <source-install>` of NEST (i.e. the
```
Comment (jessica-mitchell) [inline doc/userdoc/documentation_workflow/user_documentation_workflow.rst:86]:
```suggestion
1. Go to the :ref:`build directory <source-install>` of NEST (i.e., the
```

Comment (jessica-mitchell) [inline doc/userdoc/documentation_workflow/user_documentation_workflow.rst:86]:
```suggestion
1. Go to the :ref:`build directory <source-install>` of NEST (i.e., the
```

**Review by jessica-mitchell — APPROVED** [review]
Thanks @sarakonradi , just have one little comma to add.

**Comment by heplesser** [comment]
Minor workflow documentation improvement, therefore merging based on single review.

**Pull request merged**